sklearn/decomposition/factor_analysis.py

```diff
@@ -347,3 +359,38 @@ def score(self, X, y=None):
             Average log-likelihood of the samples under the current model
         """
         return np.mean(self.score_samples(X))
+
+    def _rotate(self, method="varimax", n_components=None, tol=1e-6):
+        "Rotate the factor analysis solution."
+        implemented = {"varimax", "quartimax"}
+        if method in implemented:
+            self.components_ = self._ortho_rotation(
```

Comment:
It seems strange to put this in a separate function. Instead:

```py
if method is None:
    return
if method not in implemented:
    raise ...
do the rotation
self.components_ = ...
```

Reply:
To make it simpler to test, you could make this a module-level private function or a staticmethod, taking `components_` in and returning it optionally modified.

Reply:
Ok sure. I'd for now keep it as a separate function to make it easier to refactor if somebody decides to add oblique or other rotations ..?